src/main/java/com/google/cloud/tools/appengine/api/devserver/DefaultRunConfiguration.java

```diff
@@ -254,4 +257,28 @@ public String getJavaHomeDir() {
   public void setJavaHomeDir(String javaHomeDir) {
     this.javaHomeDir = javaHomeDir;
   }
+
+  @Override
+  public ImmutableMap<String, String> getEnvironmentVariables() {
+    return environment;
+  }
+ 
+  /**
+   * Makes an immutable copy of the specified map. 
+   * 
+   * @param environment the environment variables to store.
+   * @throws NullPointerException if any key or value in environment is null
+   * @throws IllegalArgumentException if any key contains the NUL or = character 
+   */
+  public void setEnvironmentVariables(Map<String, String> environment) {
+    for (String key : environment.keySet()) {
+      if (key.indexOf('\0') > -1) {
```

Comment:
It's one of two characters not allowed in Unix environment variable names, the other being =